### DISCUSSION TITLE
date in yaml frontmatter from an evaluated R expression?

### DISCUSSION THREAD

**Discussion by rkrug** [quarto-dev/quarto-cli#452] [Q&A] [answered]
Hi
I use regularly in my Rmarkdown documents

```
date: "`r Sys.time()`"
```

But this results in a quarto document in an 

```
PUBLISHED
Invalid Date
```

How can I use the current date (and possibly also the tome) for the quarto document in the frontmatter?

Thanks

**Comment by dragonstyle** [comment]
Hi -  we've just been doing some work in this area, so great timing! What version of Quarto are you using? Recent versions should support that syntax for documents. 

In addition, the very latest version also supports either `today` or `last-modified` which provide the current day or the last modified date of the input qmd respectively. e.g.:

```yaml
--- 
date: today
---
```

or 

```yaml
--- 
date: last-modified
---
```

**Reply by rkrug** [reply]
Brilliant - I love it that problems are solved before they bother me...

I am using quarto 0.9.123, installed using home-brew.

**Comment by rkrug** [comment, marked as answer]
Nice - upgraded to 0.90128 and `today` and `last-updated` are working. 

One follow up question: would it be possible to also support to have the time in there?

**Reply by dragonstyle** [reply]
There may be two things in play. For 'today' we currently zero the time (assuming users likely typically mean this day). For last-modified we do preserve the time. It may just be the formatter that is omitting the time. If you'd like you can confirm by using the `date-format:` key to tinker with the format. We just added news docs for this here:

https://quarto.org/docs/reference/date-format.html

If you think a version of 'today' with a timestamp would be beneficial, we could consider also adding support for 'now' which would include a timestamp...

**Reply by rkrug** [reply]
It works with setting the date format:

```
---
date: last-modified 
date-format: "DD MMM YYYY, HH:mm"000
---
```

But I think a "now" timestamp would be beneficial, as in the case of for example reports of data processing, the time is useful when e.g. data is generated in hourly intervalls.

Thanks a lot,

Rainer

**Reply by eyayaw** [reply]
```yaml
---
date: today
date-format: long #"MMMM DD, YYYY" # "`r format(Sys.Date(), '%B %d, %Y')`"
---
```
Does not seem to work for `format: pdf`.  How does dynamic date work in pdf output? 
```
> quarto version: quarto-0.9.228-macos.pkg (Pkg)

**Reply by cderv** [reply]
@Eyayaw I can reproduce. Let's open an issue for this.  https://github.com/quarto-dev/quarto-cli/issues/615

**Comment by rkrug** [comment]
Why can't I mark @dragonstyle  reply as the answer?

**Reply by dragonstyle** [reply]
Ok, we'll add a case for 'now' as well- thanks for the feedback! I'm not sure why - perhaps you can only mark the root answers vs replies...
